demo/Dockerfile

```diff
@@ -1,10 +1,15 @@
-FROM jenkins/jenkins:2.375.2
+FROM maven:3.9.6-eclipse-temurin-11 AS maven
+FROM jenkins/jenkins:2.387.3
```

Comment:
Should stay consistent with `pom.xml` > `jenkins.version` but we don't have any automated way to enforce it.